wrappers/src/shared_realm_cs.cpp

```diff
@@ -52,92 +53,79 @@ class CSharpBindingContext: public BindingContext {
-    char* object_type;
-    bool is_nullable;
-    bool is_primary;
-    bool is_indexed;
-};
-
-struct SchemaObject
-{
-    char* name;
-    int properties_start;
-    int properties_end;
-};
-
-static util::Optional<Schema> create_schema(SchemaObject* objects, int objects_length, SchemaProperty* properties)
-{
-    std::vector<ObjectSchema> object_schemas;
-    object_schemas.reserve(objects_length);
-    
-    for (int i = 0; i < objects_length; i++) {
-        SchemaObject& object = objects[i];
-        
-        ObjectSchema o;
-        o.name = object.name;
-        
-        for (int n = object.properties_start; n < object.properties_end; n++) {
-            SchemaProperty& property = properties[n];
-            
-            Property p;
-            p.name = property.name;
-            p.type = property.type;
-            p.object_type = property.object_type ? property.object_type : "";
-            p.is_nullable = property.is_nullable;
-            p.is_indexed = property.is_indexed;
-            
-            if ((p.is_primary = property.is_primary)) {
-                o.primary_key = p.name;
-            }
-            
-            o.persisted_properties.push_back(std::move(p));
-        }
-        
-        object_schemas.push_back(std::move(o));
-    }
-    
-    return util::Optional<Schema>(std::move(object_schemas));
-}
-    
 }
 }
 
 extern "C" {
     
-    REALM_EXPORT void register_notify_realm_changed(NotifyRealmChangedT notifier)
+REALM_EXPORT void register_notify_realm_changed(NotifyRealmChangedT notifier)
 {
     notify_realm_changed = notifier;
 }
 
-REALM_EXPORT SharedRealm* shared_realm_open(uint16_t* path, size_t path_len, bool read_only, SharedGroup::DurabilityLevel durability,
-                        uint8_t* encryption_key, SchemaObject* objects, int objects_length, SchemaProperty* properties, bool delete_if_migration_needed, uint64_t schemaVersion, NativeException::Marshallable& ex)
+struct Configuration
 {
-    return handle_errors(ex, [=]() {
-        Utf16StringAccessor pathStr(path, path_len);
+    uint16_t* path;
+    size_t path_len;
+    
+    bool read_only;
+    
+    bool in_memory;
+    
+    bool delete_if_migration_needed;
+
+    uint64_t schema_version;
+    
+     bool (*migration_callback)(SharedRealm* old_realm, SharedRealm* new_realm, SchemaForMarshaling, uint64_t schema_version, void* managed_migration_handle);
+    void* managed_migration_handle;
+};
+    
+REALM_EXPORT SharedRealm* shared_realm_open(Configuration configuration, SchemaObject* objects, int objects_length, SchemaProperty* properties, uint8_t* encryption_key, NativeException::Marshallable& ex)
+{
+    return handle_errors(ex, [&]() {
+        Utf16StringAccessor pathStr(configuration.path, configuration.path_len);
 
         Realm::Config config;
         config.path = pathStr.to_string();
-        config.in_memory = durability != SharedGroup::durability_Full;
+        config.in_memory = configuration.in_memory;
```

Comment:
We should also have a unit test to confirm we can create an in-memory Realm, now we are passing this setting through.